src/PixiEditor/Models/IO/PaletteParsers/CorelDrawPalParser.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System.IO;
+using System.Text;
+using PixiEditor.Extensions.Palettes;
+using PixiEditor.Extensions.Palettes.Parsers;
+
+namespace PixiEditor.Models.IO.PaletteParsers;
+
+internal class CorelDrawPalParser : PaletteFileParser
+{
+    public override string FileName { get; } = "CorelDRAW! 3.0 Palette";
+    public override string[] SupportedFileExtensions { get; } = { ".pal" };
+
+    // Default name to use for colors (color name is required by format)
+    private const string SWATCH_NAME = "PixiEditor Color";
+
+    public override async Task<PaletteFileData> Parse(string path)
+    {
+        try
+        {
+            return await ParseFile(path);
+        }
+        catch
+        {
+            return PaletteFileData.Corrupted;
+        }
+    }
+
+    private async Task<PaletteFileData> ParseFile(string path)
+    {
+        string name = Path.GetFileNameWithoutExtension(path);
+
+        List<PaletteColor> colors = new();
+
+        await using (Stream stream = File.OpenRead(path))
+        {
+            using (TextReader reader = new StreamReader(stream, Encoding.ASCII))
+            {
+                string line;
+
+                while ((line = reader.ReadLine()) != null && (line.Length == 0 || line[0] != (char)26))
+                {
+                    if (!string.IsNullOrEmpty(line))
+                    {
+                        int lastQuote;
+                        int numberPosition;
+                        byte cyan;
+                        byte magenta;
+                        byte yellow;
+                        byte black;
+
+                        lastQuote = line.LastIndexOf('"');
+
+                        if (lastQuote == -1)
+                        {
+                            return PaletteFileData.Corrupted; // Unable to parse line
+                        }
+
+                        numberPosition = lastQuote + 1;
+                        cyan = this.NextNumber(line, ref numberPosition);
+                        magenta = this.NextNumber(line, ref numberPosition);
+                        yellow = this.NextNumber(line, ref numberPosition);
+                        black = this.NextNumber(line, ref numberPosition);
+
+                        colors.Add(this.ConvertCmykToRgb(cyan, magenta, yellow, black));
+                    }
+                }
+            }
+        }
+
+        return new PaletteFileData(name, colors.ToArray());
+    }
+
+    public override bool CanSave => false;
+
+    // NOTE: saving data works but PixiEditor defaults to first .pal parser which
+    //       is JASC so saving to an actual file is currently not available.
+    //       Palette exporter needs to be refactored to support multiple formats of
+    //       same file extension.
```

Comment:
Due to this, I can't really merge your PR. Some of core contributors would be forced to actually implement this before next release, so stuff doesn't break. If you don't implement this yourself, then this PR will be on hold for unknown amount of time (until we add this feature)

Reply:
I figured, that is why I set the save disabled attribute. The write code was let in place for future use.